train.py

```diff
@@ -169,8 +252,21 @@ def create_datasets(tokenizer, args):
         seq_length=args.seq_length,
         chars_per_token=chars_per_token,
         content_field=args.data_column,
+        fim_rate=args.fim_rate,
+        fim_spm_rate=args.fim_spm_rate
     )
-    return train_dataset, valid_dataset
+
+    valid_dataset_no_fim = ConstantLengthDataset(
```

Comment:
Did you want to add an option to compute validation loss in non FIM mode? Currently this dataset is not used in the script